src/java/org/apache/cassandra/db/RangeTombstoneList.java

```diff
@@ -121,9 +122,11 @@ public void add(RangeTombstone tombstone)
      */
     public void add(ByteBuffer start, ByteBuffer end, long markedAt, int delTime)
     {
+        assert(comparator.compare(start, end) <=0);
         if (isEmpty())
         {
             addInternal(0, start, end, markedAt, delTime);
+            assertSorted();
```

Comment:
Question: Should we have assertSorted here since we know the current list is empty anyways? is there a chance that things would get out of order even if the current list is empty?